File: exception_handler/csharp_exception_handler/exception_handler.cs
using System;

public class ExceptionHandler {
  public static void Main( string[] args ) {
    if( args.Length != 0 ) {
      Console.Error.WriteLine( "Usage: " + Environment.GetCommandLineArgs()[0] + " <noargs>" );
      Environment.Exit(1);
    }

    for( int idx_i = 0; idx_i < 10; idx_i++ ) {
      Console.WriteLine("example_function(" + idx_i + ")=" + example_function(idx_i));
      Console.WriteLine("example_function()=" + example_function());
    }
  }
  public static int example_function( int n ) {
    try {
      n = 1/(n-n);
    }
    catch( DivideByZeroException dze ) {
      Console.WriteLine(dze.Message);
    }
    catch( Exception e ) {
      Console.WriteLine(e);
      Console.WriteLine(e.Message);
    }
    finally {
      Console.WriteLine("finally");
    }
    return n;
  }

  public static int example_function() {
    try {
      if( true ) {
        throw new Exception();
      }
    }
    catch( Exception e ) {
      Console.WriteLine("WARN: example_function()");
      Console.WriteLine(e);
    }
    return 0;
  }
}

File: exception_handler/java_exception_handler/ExceptionHandler.java
import java.lang.*;
//comment

public class ExceptionHandler {
  public ExceptionHandler() {
  }

  public static int example_function() {
    try {
      if( true ) {
        throw new Exception();
      }
    }
    catch( Exception e ) {
      System.out.println("WARN: example_function()");
      System.out.println(e);
    }
    return 0;
  }

  public static int example_function( int n ) {
    try {
      n = 1/0;
    }
    catch( ArithmeticException ae ) {
      System.out.println("WARN: ArithmeticException");
      System.out.println(ae);
    }
    catch( Exception e ) {
      System.out.println(e);
    }
    finally {
      System.out.println("finally called");
      return n;
    }
  }

  public static void main( String[] args ) {
    if( args.length != 0 ) {
      System.err.println( "Usage: java -jar ExceptionHandler.jar <noargs>");
      System.exit(1);
    }
    for( int idx_i = 0; idx_i < 10; idx_i++ ) {
      System.out.println("example_function(" + idx_i + ")=" + example_function(idx_i));
      System.out.println("example_function()=" + example_function());
    }
  }
}
